File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> map = new HashMap<>();
        for(String str: participant){
            map.put(str, map.getOrDefault(str,0)+1);
        }
        int x;
        for(String str: completion){
            x = map.get(str);
            if(x==1) map.remove(str);
            else map.put(str, x-1);
        }
        for(Map.Entry<String, Integer> entry : map.entrySet())
            answer= entry.getKey();
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import defaultdict

def solution(participant, completion):
    answer = ''
    part_cnt = defaultdict(int)
    for ppl in participant:
        part_cnt[ppl]+=1
    for ppl in completion:
        if part_cnt[ppl]==1:
            part_cnt.pop(ppl)
        else : part_cnt[ppl]-=1
    for ppl in part_cnt.keys():
        answer = ppl
    return answer